src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -185,6 +185,30 @@ impl LocationState {
                 // need to be applied to this subtree.
                 _ => false,
             };
+            if self.permission.is_disabled() {
+                // A foreign access to a `Disabled` tag will have almost no observable effect.
+                // It's a theorem that `Disabled` node have no protected initialized children,
```

Comment:
That's not an obvious theorem -- can you give a brief argument?

Reply:
It's proven in Coq :stuck_out_tongue:.

The reason it holds is that to become disabled, you need to have a foreign write access happen. But that would have triggered any protected initialized nodes that are children of the node being disabled. And you can't have a new child of Disabled become initialized, because that would mean the to-be-initialized node has a child access, which is however blocked by the Disabled parent.